create content guidelines (#877)
SAP/luigi: docs/content-guidelines.md

# Content Guidelines

This page contains instructions on how to create documentation for Luigi. It defines the rules for each of these topics:
* [Structure](#structure)
* [Metadata](#metadata)
* [Audience and language](#audience-and-language)
* [Format](#format)
* [Links](#links)
* [Screenshots and diagrams](#screenshots-and-diagrams)
* [Glossary](#glossary)

## Overview

Luigi documentation exists in the form of Markdown files on GitHub, which are then transferred to the main documentation page using Sapper. Find more about what GitHub-flavored Markdown is [here](https://github.github.com/gfm).

> Documentation is located in the `luigi/docs` folder in the [Luigi repository](https://github.com/SAP/luigi).

Reviewer: ```suggestion
Documentation is located in the `luigi/docs` folder in the [Luigi repository](https://github.com/SAP/luigi).
```
it can also be `documentation resides` if you want to keep the active voice. But `is located` is also good :)

- Use active voice. For example, instead of writing "Luigi Client should be installed...", write "Install Luigi Client..."
- Do not use slang or abbreviations. This also means you should not use contractions ("don't" instead of "do not") or short forms ("info" instead of "information").
- Use the present tense.
- Use concise language and avoid long blocks of text. Lists, tables, or subheadings can help you with that.
- Give practical examples of features instead of only using words to explain them. Additionally, link to the [Luigi Fiddle](https://fiddle.luigi-project.io/) as a tool where users can experiment with features.

## Format

This section provides you with guidelines on how to format and organize your text.

### Headings

Use H1 headings (preceded by # in Markdown) only at the start of the document to indicate the document name.

Only use H2 and H3 headings to organize your content into categories. Do not use H4 (####) headings or lower.

### Lists

Lists are very useful for breaking up text and providing instructions.

- Use bullet points (created with * or -) for lists involving general explanations.
- Use numbered lists only for step-by-step instructions.

### Tables

> Use tables when content needs comparison or to organize small bits of information. Keep in mind that in Markdown it is not possible to break up the content of a table cell into separate lines or bullet points, therefore tables are not useful when they contain long sentences or lists.

Reviewer: ```suggestion
Use tables when content needs comparison or to organize small bits of information. Keep in mind that in Markdown it is not possible to break up the content of a table cell into separate lines or bullet points, therefore tables are not useful when they contain long sentences or lists.
```
You actually can break it in separate lines using `<br>`  or some other tricks: https://gist.github.com/shaunlebron/746476e6e7a4d698b373 :D I'm not saying we necessary advise that, but it is possible and sometimes I do use it (when I need to list more terms). 
But the general comment that you shouldn't add long lines is good :)

You can find an example of a table in the [Styles](#styles) section of this document.

### Code snippets

> Code snippets must be surrounded with the Markdown code block tag and the programming language should be specified. Make sure to indent code correctly using your text editor (2 space indentation is the default).

Reviewer: ```suggestion
Surround code snippets with the Markdown code block tag and specify the programming language. Make sure to indent code correctly using your text editor (2 space indentation is the default).
```
You can also use the active voice here.

For example:

```javascript
Luigi.setConfig({
  routing: {
    nodeParamPrefix: '~'
  },
```

is better than:

```
Luigi.setConfig({
    routing: {
        nodeParamPrefix: '~'
    },
```

### Styles

> Depending on the type of content, you should use different types of text, for example **bolded** or `code`.

Reviewer: ```suggestion
Depending on the type of content,  use different types of text, for example, **bolded** or `code`.
```

|     Type           |      Font          |    Example    |
|--------------------|:------------------:|---------------|
| Parameters         |      **bold**     | **viewGroup** |
| Attributes | **bold** | **collapsible** |
| Values         |      `code`     | `true`, `false` |
| Dynamic parameters         |      `code`        | `{userId}`    |
| Folders, paths, filenames  |      `code`        | Open `basicConfiguration.js` inside `assets/luigi-config`|
| Code snippets      |      `code`        | See [this section](#code-snippets) |
| Functions          |      `code`     | `showLoadingIndicator()` |

## Links

> If the link is within the same folder on GitHub, use only the relative path. For all other links, including external links, use the absolute path (which starts with https://...).

Reviewer: ```suggestion
If the link is within the same folder on GitHub, use only the relative path. For all other links, including external links, use the absolute path, starting with `https://`.
```

To link to a section within a document, use the title of that section as an anchor. For example:

```
[Link to the "custom messages" section in the "communication" document](communication.md#custom-messages)
```

## Screenshots and diagrams

> When adding screenshots or diagrams you should adhere to to the following rules:

Reviewer: ```suggestion
When adding screenshots or diagrams, adhere to the following rules:
```

> - Only use high resolution images

Reviewer: ```suggestion
- Only use high-resolution images.
```

> - Only use screenshots or diagrams if necessary, as too many of them can create visual noise

Reviewer: ```suggestion
- Only use screenshots or diagrams if necessary, as too many of them can create visual noise.
```

> - As an alternative to screenshots, point to the [Luigi fiddle](https://fiddle.luigi-project.io/) or other Luigi examples when you want to illustrate a concept.

Reviewer: ```suggestion
- As an alternative to screenshots, point to the [Luigi fiddle](https://fiddle.luigi-project.io/) or other Luigi examples when you want to illustrate a concept.
```
is it `luigi fiddle` or `the luigi fiddle`?

Author: I changed it to "Luigi Fiddle"

Author: I changed it to "Luigi Fiddle"

## Glossary

This section contains terminology frequently used in the Luigi documentation.

* **Luigi Core** - the main application and the settings used to configure it.
* **Luigi Client** - the micro frontend within Luigi Core and the API used to connect the two.
* **Luigi Fiddle** - a [page](https://fiddle.luigi-project.io/) where you can configure an example Luigi application and explore Luigi functions.
* **Parameters** - the parameters that can be used to configure Luigi in the Luigi configuration file, for example **category**, **viewUrl**, and more.
* **Dynamic parameters** - the parameters that can be added to create a dynamically changeable path.

> * **Attributes** - the "sub properties" of parameters. E.g. the **category** parameter can have **label**, **icon**, and **collapsible** as attributes.

Reviewer: ```suggestion
* **Attributes** - the "sub properties" of parameters. For example, the **category** parameter can have **label**, **icon**, and **collapsible** as attributes.
```
